src/db/migrations/43-add-lens.ts

```diff
@@ -0,0 +1,182 @@
+import { Knex } from 'knex';
+
+import { formatAddress } from '../../types/address';
+import { Chain, ChainId, ChainType } from '../../types/chain';
+import { getCrdtUpsertMessage } from '../../types/message';
+import { MetaFactory, MetaFactoryTypeName } from '../../types/metaFactory';
+import { NFTStandard } from '../../types/nft';
+import { MusicPlatform, MusicPlatformType } from '../../types/platform';
+import { Table } from '../db';
+import { updateViews } from '../migration-helpers';
+
+const LENS_PLATFORM: MusicPlatform = {
+  id: 'lens',
+  type: MusicPlatformType['multi-track-multiprint-contract'],
+  name: 'LENS'
+}
+
+const LENS_HUB: MetaFactory = {
+  id: `${ChainId.polygon}/0xDb46d1Dc155634FbC732f92E853b10B288AD5a1d`,
+  platformId: LENS_PLATFORM.id,
+  contractType: MetaFactoryTypeName.lens,
+  standard: NFTStandard.ERC721,
+  autoApprove: false,
+  startingBlock: '28384640',
+  chainId: ChainId.polygon,
+  address: formatAddress('0xDb46d1Dc155634FbC732f92E853b10B288AD5a1d')
+}
+
+const ETHEREUM: Chain = {
+  id: ChainId.ethereum,
+  name: 'Ethereum',
+  type: ChainType.evm,
+  rpcUrl: process.env.ETHEREUM_PROVIDER_ENDPOINT!
+}
+
+const POLYGON: Chain = {
+  id: ChainId.polygon,
+  name: 'Polygon POS',
+  type: ChainType.evm,
+  rpcUrl: process.env.POLYGON_PROVIDER_ENDPOINT!
+}
+
+const SOLANA: Chain = {
+  id: ChainId.solana,
+  name: 'Solana',
+  type: ChainType.solana,
+  rpcUrl: process.env.SOLANA_PROVIDER_ENDPOINT!
+}
+
+export const up = async (knex: Knex) => {
+
+  // create chains
+  await knex.schema.createTable(Table.chains, table => {
+    table.string('id').primary(),
+    table.string('name').notNullable(),
+    table.string('rpcUrl').notNullable()
+    table.string('type').notNullable()
+  })
+
+  await knex(Table.chains).insert(ETHEREUM);
+  await knex(Table.chains).insert(POLYGON);
+  await knex(Table.chains).insert(SOLANA);
+
+  // alter factories
+  await knex.schema.alterTable(Table.nftFactories, table => {
+    table.string('chainId')
+    table.string('address')
+  })
+  
+  await knex.schema.alterTable(Table.metaFactories, table => {
+    table.string('chainId')
+    table.string('address')
+  })
+
+  // update factories
```

Comment:
do we also need an update to update existing nftFactories.address and metaFactories.address? maybe can do in follow up PR if needed tho

Reply:
This migration already updates nftFactories.address and metaFactories.address:

```const nftFactoryUpdates = "
    update ${Table.nftFactories}
    set address=id,
    "chainId"= 
    case
      when "standard" = '${NFTStandard.METAPLEX}' then 'solana'
      else 'ethereum'
    end
  "
  const metaFactoryUpdates = "
    update ${Table.metaFactories}
    set address=id,
    "chainId"= 
    case
      when "standard" = '${NFTStandard.METAPLEX}' then 'solana'
      else 'ethereum'
    end
  "
```